File: problems/algorithms/regexp-pattern-validator/js/response/pattern-validator.js
import { checkingDigitCpf } from '../../../string/validating-cpf/response/validating-cpf.js';

function validateCpf(cpf) {
  return (
    /^(\d{11}|(\d{3}\.){2}\d{3}-\d{2})$/.test(cpf) && checkingDigitCpf(cpf.replace(/[\.-]/g, ''))
  );
}

function validateCnpj(cnpj) {
  // TODO
  // return /^()$/.test(cnpj)
}

function validateCep(cep) {
  return /^(\d{8}|\d{2}[\.-]?\d{3}-\d{3})$/.test(cep);
}

function validateOctal(octal) {
  return /^0[oO]?[0-7]+$/.test(octal);
}

export { validateCpf, validateCnpj, validateCep, validateOctal };

File: problems/algorithms/regexp-pattern-validator/ruby/response/src/pattern_validator.rb
class PatternValidator

  def PatternValidator.validateCpf()
    /^([0-9]{11})|([0-9]{3}[\.][0-9]{3}[\.][0-9]{3}-[0-9]{2})$/
  end

  def PatternValidator.validateCep()
    /^([0-9]{8})|([0-9]{5}-[0-9]{3})$/
  end

end
